core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -54,7 +54,29 @@ enum SchemaVersion {
   static final String TABLE_RECORD_TYPE = "TABLE";
   static final String VIEW_RECORD_TYPE = "VIEW";
 
-  private static final String V1_DO_COMMIT_SQL =
+  private static final String V1_DO_COMMIT_TABLE_SQL =
+      "UPDATE "
+          + CATALOG_TABLE_VIEW_NAME
+          + " SET "
+          + JdbcTableOperations.METADATA_LOCATION_PROP
+          + " = ? , "
+          + JdbcTableOperations.PREVIOUS_METADATA_LOCATION_PROP
+          + " = ?"
+          + " WHERE "
+          + CATALOG_NAME
+          + " = ? AND "
+          + TABLE_NAMESPACE
+          + " = ? AND "
+          + TABLE_NAME
+          + " = ? AND "
+          + JdbcTableOperations.METADATA_LOCATION_PROP
+          + " = ? AND ("
+          + RECORD_TYPE
+          + " = ?"
```

Comment:
Should we hardcode the record type to TABLE since it is only called for table?
Similar comment for the `V1_DO_COMMIT_VIEW_SQL`.

or may be we just have to add "OR RECORD_TYPE IS NULL" to previous `V1_DO_COMMIT_SQL` without new changes.

Reply:
Yes I can hardcode. Just adding the or clause won't work as we need the parenthesis.